source/utf8/unchecked.h

```diff
@@ -97,6 +97,37 @@ namespace utf8
             return utf8::unchecked::replace_invalid(start, end, out, replacement_marker);
         }
 
+        template <typename octet_iterator>
+        uint32_t peek(octet_iterator it)
```

Comment:
I feel this duplicates logic already implemented somewhere (I guess in `next`) and should be avoided

Reply:
There is already `peek_next`. Maybe just fix and use that?